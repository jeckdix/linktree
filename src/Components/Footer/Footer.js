import React from "react";
import styled from "styled-components";

const FooterSection = styled.footer`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 10px 0px;
  background: #ffffff;  
  @media (min-width: 1200px) {
    padding: 32px 0px;

  }

`;
const Container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: left;
  align-items: left;
  padding: 2px 10px;
  gap: 2px;

  @media (min-width: 1200px) {
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    width: 1280px;
    padding: 2px 0px;

  }
`;

const StyledImg = styled.img`
  width: 30%;
  @media (min-width: 1200px) {
    width: 10%;
  }
`;
const StyledText = styled.h3`
  font-family: "Inter";
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 24px;

  color: #667085;
  @media (min-width: 1200px) {
    font-size: 16px;
    width: 380px;
    text-align: left;
  }
`;


const ZuriSvg = styled.svg `
height: 20px;
width: 300px;
font-size: 20px;
font-weight: 550;
word-spacing: 3px;
font-family: Arial, Helvetica, sans-serif;
`

const Footer = () => {
  return (
    <FooterSection>
      <Container>
        <ZuriSvg>
          <text x="0" y="15" fill="black" >Zuri  Internship </text>
          <circle cx="42" cy="12" r="3" fill="red" />
        </ZuriSvg>
        <StyledText>HNG Internship 9 Frontend Task</StyledText> 
        <StyledImg src={require("../../Assets/I4G.png")} alt="I4G" />
      </Container>
    </FooterSection>
  );
};

export default Footer;

