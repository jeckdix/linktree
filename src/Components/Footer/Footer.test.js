import React from 'react'
import { render, screen, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'
import Footer from './Footer'

describe('Footer', () => {
    afterAll(() => {
        cleanup();
    });

    it('should render without errors', () => {

        const { asFragment } = render(<Footer  />);
        expect(asFragment()).toMatchSnapshot();
    });


    it('should a footer text', () => {

        render(<Footer  />);
        expect(screen.getByText("HNG Internship 9 Frontend Task")).toBeInTheDocument();
    });

    // it('should contain the correct attributes', () => {
    //         const linkProps = {
    //                 id: "btn__zuri",
    //                 link: "https://training.zuri.team/",
    //                 title: "Zuri Team",
    //         }
    //         render(<Button {...linkProps} />);

    //         expect(screen.getByRole('link')).toHaveAttribute('href', linkProps.link);
    //         expect(screen.getByRole('link')).toHaveAttribute('id', linkProps.id);
    //         expect(screen.getByRole('link')).toHaveTextContent(linkProps.title);
    // })
})

// jestdom
// react-testing-library



