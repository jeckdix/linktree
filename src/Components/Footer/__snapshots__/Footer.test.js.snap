// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render without errors 1`] = `
<DocumentFragment>
  <footer
    class="sc-bcXHqe dfQTNY"
  >
    <div
      class="sc-gswNZR jkUtxa"
    >
      <img
        alt="footer"
        class="sc-dkrFOg ehcWUz"
        src="Zuri-internship.png"
      />
      <h3
        class="sc-hLBbgP fynkll"
      >
        HNG Internship 9 Frontend Task
      </h3>
      <img
        alt="I4G"
        class="sc-dkrFOg ehcWUz"
        src="I4G.png"
      />
    </div>
  </footer>
</DocumentFragment>
`;
